File: packages/swift-ui/Sources/OrotpSwiftUI/VerificationStatus.swift
import Foundation

/// The five lifecycle states a verification can be in. The wire
/// format spells them in lowercase; Swift's default `String`
/// rawValue for `case foo` is `"foo"` — which matches the wire
/// format byte-for-byte.
public enum VerificationStatus: String, Codable, Sendable, CaseIterable {
    case pending
    case verified
    case expired
    case cancelled
    case failed

    public var isTerminal: Bool { self != .pending }
}

File: packages/android-ui/library/src/main/kotlin/dev/syrotp/ui/VerificationStatus.kt
package dev.syrotp.ui

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * The five lifecycle states a verification can be in. The wire
 * format spells them in lowercase; `@SerialName` keeps the Kotlin
 * idiom (PascalCase enum constants) without changing the JSON
 * shape your backend hands you.
 */
@Serializable
enum class VerificationStatus {
    @SerialName("pending") Pending,
    @SerialName("verified") Verified,
    @SerialName("expired") Expired,
    @SerialName("cancelled") Cancelled,
    @SerialName("failed") Failed;

    val isTerminal: Boolean get() = this != Pending
}
